micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -85,4 +85,17 @@ public void close() {
         stop();
         super.close();
     }
+
+    private static void warn(String message, Throwable t) {
+        try {
+            LoggerFactory.getLogger(PushMeterRegistry.class).warn(message, t);
+        }
+        catch (Throwable ex) {
+            StringWriter writer = new StringWriter();
+            t.printStackTrace(new PrintWriter(writer));
+            String stackTrace = writer.toString();
+            System.out.println(message + System.lineSeparator() + stackTrace);
```

Comment:
Do we want to just go to no-op here? JUL?

Reply:
@jkschneider Thanks for the feedback!

Looking at the [Netty's implementation for internal loggers](https://github.com/netty/netty/blob/78adeb540833251c3bf31b5c2d92cb3343c84461/common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java#L41-L61), it has the following precedence for logging frameworks: SLF4J > Log4J > Log4J2 > JUL

I think we can have a simplified version of it for now, for example: SLF4J > JUL

Please let me know which direction you prefer. Then I'll update this accordingly.

Reply:
Something like Netty's delegation to an available logging framework is preferable to `System.out.println`, I think. Let's go that direction.

Reply:
Also starting with a simplified version that does SLF4J > JUL as suggested sounds good to me. Do you want to update this pull request with that, or shall we make a separate issue to track it, @izeye ?

Reply:
@shakuzen Thanks for the feedback! I'd like to copy and paste the code for internal logging in Netty rather than reinvent the wheel as it seems to be complete and well written already. IANAL, but both Netty and Micrometer are licensed under the Apache License, Version 2.0, so it seems okay. Is it okay to go this route?

Reply:
IANAL either, but I believe that's okay. As a courtesy and for attribution, we can mention in the JavaDocs that it was taken from the Netty project.

Reply:
@shakuzen Thanks for the quick feedback 👍 I'll update this PR with a simplified version (SLF4J > JUL) then.